### ISSUE TITLE
Training speed much slower

### ISSUE THREAD

**Issue by frunkad** [tensorflow/tensorflow#20550]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
The code used is adapted from [Keras FastText](https://github.com/keras-team/keras/blob/master/examples/imdb_fasttext.py)  || [Exact Code](https://github.com/frunkad/sent-aog/blob/master/imdb_fasttext.py)
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:Linux Ubuntu 18.04 
- **TensorFlow installed from (source or binary)**: Anaconda
- **Bacel Version**: Not applicable
- **TensorFlow version**: v1.8.0-0-g93bc2e2072 1.8.0
- **Python version**: Python 3.6.4 :: Anaconda custom (64-bit)
- **CUDA/cuDNN version**: 9.0, V9.0.176 
- **GPU model and memory**: GeForce GTX 980M totalMemory: 7.93GiB
- **Exact command to reproduce**: device specific

From Environment Capture Script:
```
== cat /etc/issue ===============================================
Linux dracarys 4.15.0-23-generic #25-Ubuntu SMP Wed May 23 18:02:16 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
VERSION="18.04 LTS (Bionic Beaver)"
VERSION_ID="18.04"
VERSION_CODENAME=bionic

== are we in docker =============================================
No

== compiler =====================================================
c++ (Ubuntu 7.3.0-16ubuntu3) 7.3.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


== uname -a =====================================================
Linux dracarys 4.15.0-23-generic #25-Ubuntu SMP Wed May 23 18:02:16 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

== check pips ===================================================
numpy                              1.14.4     
numpydoc                           0.7.0      
protobuf                           3.5.2.post1
tensorflow-gpu                     1.8.0      

== check for virtualenv =========================================
False

== tensorflow import ============================================
tf.VERSION = 1.8.0
tf.GIT_VERSION = v1.8.0-0-g93bc2e2072
tf.COMPILER_VERSION = v1.8.0-0-g93bc2e2072
Sanity check: array([1], dtype=int32)
/home/frunkad/anaconda3/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters

== env ==========================================================
LD_LIBRARY_PATH /usr/local/cuda/lib64:
DYLD_LIBRARY_PATH is unset

== nvidia-smi ===================================================
Thu Jul  5 02:05:17 2018       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 390.48                 Driver Version: 390.48                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 980M    Off  | 00000000:01:00.0  On |                  N/A |
| N/A   48C    P0    31W /  N/A |    588MiB /  8118MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      1530      G   /usr/lib/xorg/Xorg                            16MiB |
|    0      1579      G   /usr/bin/gnome-shell                          48MiB |
|    0      1762      G   /usr/lib/xorg/Xorg                           212MiB |
|    0      1907      G   /usr/bin/gnome-shell                          88MiB |
|    0      2394      G   ...-token=8EBF6E3FCA82B57AEE9186BDD7AFEAF3   108MiB |
|    0      2902      G   ...-token=FA087B07103596ED40954ABE2071373B   105MiB |
+-----------------------------------------------------------------------------+

== cuda libs  ===================================================
/usr/local/cuda-9.0/lib64/libcudart_static.a
/usr/local/cuda-9.0/lib64/libcudart.so.9.0.176
/usr/local/cuda-9.0/doc/man/man7/libcudart.7
/usr/local/cuda-9.0/doc/man/man7/libcudart.so.7

```
### Describe the problem
From the keras code the GTX 980M should have taken 2s for each epoch, but it takes around 58s. Also the same code runs within 10s on colab. The possible reason as per me is that I am using it with Ubuntu 18. Should I give it a try on Windows on same machine?

### Source code / logs

```
/home/frunkad/anaconda3/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Loading data...
25000 train sequences
25000 test sequences
Average train sequence length: 238
Average test sequence length: 230
Adding 2-gram features
Average train sequence length: 476
Average test sequence length: 428
Pad sequences (samples x time)
x_train shape: (25000, 400)
x_test shape: (25000, 400)
Build model...
Train on 25000 samples, validate on 25000 samples
Epoch 1/5
2018-07-05 01:26:20.525149: I tensorflow/core/platform/cpu_feature_guard.cc:140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2018-07-05 01:26:20.601376: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:898] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2018-07-05 01:26:20.601794: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1356] Found device 0 with properties:
name: GeForce GTX 980M major: 5 minor: 2 memoryClockRate(GHz): 1.1265
pciBusID: 0000:01:00.0
totalMemory: 7.93GiB freeMemory: 7.32GiB
2018-07-05 01:26:20.601813: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1435] Adding visible gpu devices: 0
2018-07-05 01:26:20.751183: I tensorflow/core/common_runtime/gpu/gpu_device.cc:923] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-07-05 01:26:20.751214: I tensorflow/core/common_runtime/gpu/gpu_device.cc:929]      0
2018-07-05 01:26:20.751220: I tensorflow/core/common_runtime/gpu/gpu_device.cc:942] 0:   N
2018-07-05 01:26:20.751487: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1053] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 7064 MB memory)-> physical GPU (device: 0, name: GeForce GTX 980M, pci bus id: 0000:01:00.0, compute capability: 5.2)
25000/25000 [==============================] - 58s 2ms/step - loss: 0.6651 - acc: 0.6945 - val_loss: 0.6278 - val_acc: 0.7829
Epoch 2/5
25000/25000 [==============================] - 57s 2ms/step - loss: 0.5572 - acc: 0.8524 - val_loss: 0.5116 - val_acc: 0.8378
Epoch 3/5
25000/25000 [==============================] - 57s 2ms/step - loss: 0.4034 - acc: 0.9102 - val_loss: 0.4015 - val_acc: 0.8692
Epoch 4/5
25000/25000 [==============================] - 58s 2ms/step - loss: 0.2781 - acc: 0.9420 - val_loss: 0.3359 - val_acc: 0.8827
Epoch 5/5
25000/25000 [==============================] - 58s 2ms/step - loss: 0.1966 - acc: 0.9619 - val_loss: 0.2986 - val_acc: 0.8923
```

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Bazel version

**Comment by frunkad** [comment]
Bazel version: not applicable

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by bignamehyp** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?
